File: code/java/DesignPatterns/src/main/java/state/TrafficLight.java
package state;

class TrafficLight {
    private State currentState;

    public TrafficLight(State state) {
        this.currentState = state;
    }

    public void change() {
        currentState.handleRequest();
        // Logic to change the current state
    }

    public void setState(State state) {
        this.currentState = state;
    }
}

File: code/csharp/DesignPatterns/state/TrafficLight.cs
namespace State;
using System;

class TrafficLight
{
    private IState _currentState;

    public TrafficLight(IState state)
    {
        this._currentState = state;
    }

    public void Perform()
    {
        _currentState.HandleRequest();
        // Logic to change the current state
    }

    public void SetState(IState state)
    {
        this._currentState = state;
    }
}
